ZENKO-3748 - include third party dashboards in ISO

lavigne958 opened the pull request:
Include the third party dashboards that are located in
`monitoring/dashboards` in the ISO file.

This is the first step of introducing the deployment of dashboards.

The current dashboards won't be deployed as the tool
necessary to prepare the file `registry-config.inc.j2`
must be updated.

Commit:
ZENKO-3748 - include third party dashboards in ISO

Include the thrid party dashboards that are located in
`monitoring/dashboards` in the ISO file.

This is the first step of introducing the deployment of dashboards.

The current dashboards won't be deployed as the tool
necessary to prepare the file `registry-config.inc.j2`
must be updated.

bert-e commented:
# Hello lavigne958,



My role is to assist you with the merge of this
pull request. Please type `@bert-e help` to get information
on this process, or consult the [user documentation](
https://eve.devsca.com/github/scality/zenko/bert-e/doc/user).





*Status report is not available.*

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* 2 peers

lavigne958 committed:
ZENKO-3748 - use new image that can handle dashboards

francoisferrand commented on solution/build.sh:
```diff
@@ -97,6 +97,72 @@ function copy_image()
+    "layers": [
+        {
+            "mediaType": "application/grafana-dashboard+json",
+            "digest": "sha256:${digest}",
+            "size": ${size},
```
i would suggest introducing a function which generates this output from an input stream, computes the sha256sum, renames the file and generates this block:

```
# $1 : mime-type of the file
# $2 : optionally, name of the file
# input : the content of the file
# output : the manifest fragment defining the file
function processFile {
   local tmp=$(mktemp xxxxx)
   cat > $tmp
    digest=$(sha256sum  ${tmp} | cut -d " " -f 1)     # get sha256, sha256sum prints the checksum and the filename, keep the checksum only
    size=$(stat --printf "%s" ${tmp})     # get only the size (in bytes) of the file. format "%s" only prints the size
    mv $tmp $digest
    
    echo "mediaType": "$1",
    echo "digest": "sha256:${digest}",
    echo "size": ${size},
    [ "$2" ] && echo "\"annotations\": { \"org.opencontainers.image.title\": \"$2\" }"
}
```

It can then be used both the the config and the dashboard:

```
$(processFile "application/vnd.oci.image.config.v1+json" <<< '{}')
$(processFile "application/grafana-dashboard+json" "$filename" < "$file")
```

(note that this will become more useful if/when we have multiple dashboards in the same image)

lavigne958 replied:
I thought I'll do this later, for now we don't have any dashboards for others than third parties so once we are clear on the number of dashboards, if we have multiple dashboards, if we need to etc we can re-work the function that produces the manifest file. for now we don't know exactly what we are gonna need.

lavigne958 replied:
done

francoisferrand commented on solution/build.sh:
```diff
@@ -97,6 +97,72 @@ function copy_image()
+    size=$(stat --printf "%s" ${file})
+
+    # generate empty config file
+    config_file=${dashboard_base_dir}/config.json
```
should not "blindly" a (temp) file in the source code, it may conflict with an existing file (we never know...) : use mktemp instead

francoisferrand commented:
also, i'm wondering why you need to change the version on each "renamed" dashboard, except on kafka... should be the same behavior for all 🧐

francoisferrand commented on eve/main.yml:
```diff
@@ -2,7 +2,7 @@
 
 branches:
-  documentation/*, feature/*, improvement/*, bugfix/*, w/*, q/*, hotfix/*:
+  documentation/*, feature/*, improvement/*, bugfix/*, w/*, q/*, hotfix/*, user/*:
```
should this really be commited ?

lavigne958 replied:
of course it should, we forgot to have the `user/*` branches covered in this repo. I needed to add it each time I needed to run the cI. :laughing:

lavigne958 commented:
> 

I have no clue, but the script that checks the version of the dashboards says it should be version 6. so I just put the expected number :lab_coat:

Commit:
ZENKO-3748 - Use dedicated function to produce manifest layer

Use a dedicated function to produce a single layer to be stored as
JSON in the manifest file.
This will be usefull later when building manifest file for components.

lavigne958 commented:
Updated the build.sh to use the proposal from Francois.
Made a function to produce a single layer of an image.

lavigne958 commented:
/approve

bert-e commented:
# In the queue



The changeset has received all authorizations and has been added to the
relevant queue(s). The queue(s) will be merged in the target development
branch(es) as soon as builds have passed.

The changeset will be merged in:


* :heavy_check_mark: `development/2.2`



The following branches will **NOT** be impacted:

* `development/0.9`
* `development/1.0`
* `development/1.1`
* `development/1.2`
* `development/1.3`
* `development/2.1`



There is no action required on your side. You will be notified here once
the changeset has been merged. In the unlikely event that the changeset
fails permanently on the queue, a member of the admin team will
contact you to help resolve the matter.

*IMPORTANT*

Please do not attempt to modify this pull request.

* Any commit you add on the source branch will trigger a new cycle after the
  current queue is merged.
* Any commit you add on one of the integration branches will be **lost**.

If you need this pull request to be removed from the queue, please contact a
member of the admin team now.




*The following options are set:* **approve**

bert-e commented:
# 


I have successfully merged the changeset of this pull request
into targetted development branches:


* :heavy_check_mark: `development/2.2`



The following branches have **NOT** changed:

* `development/0.9`
* `development/1.0`
* `development/1.1`
* `development/1.2`
* `development/1.3`
* `development/2.1`



Please check the status of the associated issue ZENKO-3748.

Goodbye lavigne958.
